File: app/src/main/java/me/sunhapper/spcharedittool/emoji/EmojiconPagerAdapter.java
package me.sunhapper.spcharedittool.emoji;

import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;
import java.util.List;

public class EmojiconPagerAdapter extends PagerAdapter {

  private List<View> views;

  public EmojiconPagerAdapter(List<View> views) {
    this.views = views;
  }

  @Override
  public int getCount() {
    return views.size();
  }

  @Override
  public boolean isViewFromObject(View arg0, Object arg1) {
    return arg0 == arg1;
  }

  @Override
  public Object instantiateItem(ViewGroup arg0, int arg1) {
    arg0.addView(views.get(arg1));
    return views.get(arg1);
  }

  @Override
  public void destroyItem(ViewGroup arg0, int arg1, Object arg2) {
    arg0.removeView(views.get(arg1));

  }

}

File: app/src/main/java/me/sunhapper/spcharedittool/emoji/EmojiconPagerAdapter.kt
package me.sunhapper.spcharedittool.emoji

import android.support.v4.view.PagerAdapter
import android.view.View
import android.view.ViewGroup

class EmojiconPagerAdapter(private val views: List<View>) : PagerAdapter() {

    override fun getCount(): Int {
        return views.size
    }

    override fun isViewFromObject(view: View, obj: Any): Boolean {
        return view === obj
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(views[position])
        return views[position]
    }

    override fun destroyItem(container: ViewGroup, position: Int, obj: Any) {
        container.removeView(views[position])

    }

}
